apps/metadata-editor/src/app/dashboard/search-filters-summary-item/search-filters-summary-item.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+import { CommonModule } from '@angular/common'
+import {
+  FieldsService,
+  FieldType,
+  FieldValue,
+  FieldValues,
+  SearchFacade,
+  SearchService,
+} from '@geonetwork-ui/feature/search'
+import {
+  catchError,
+  filter,
+  firstValueFrom,
+  map,
+  Observable,
+  startWith,
+  switchMap,
+  tap,
+} from 'rxjs'
+import { BadgeComponent } from '@geonetwork-ui/ui/inputs'
+import { TranslateModule } from '@ngx-translate/core'
+import { DateRange } from '@geonetwork-ui/api/repository'
+
+@Component({
+  selector: 'md-editor-search-filters-summary-item',
+  standalone: true,
+  imports: [CommonModule, TranslateModule, BadgeComponent],
+  templateUrl: './search-filters-summary-item.component.html',
+  styleUrls: ['./search-filters-summary-item.component.css'],
+})
+export class SearchFiltersSummaryItemComponent implements OnInit {
+  @Input() fieldName: string
+  fieldType: FieldType
+
+  fieldValues$ = this.searchFacade.searchFilters$.pipe(
+    switchMap((filters) =>
+      this.fieldsService.readFieldValuesFromFilters(filters)
+    ),
+    tap((fieldValues) => console.log(fieldValues)),
+    map((fieldValues) =>
+      Array.isArray(fieldValues[this.fieldName]) //TODO: handle date ranges as arrays everywhere?
+        ? fieldValues[this.fieldName]
+        : [fieldValues[this.fieldName]]
+    ),
+    tap((fieldValues) => console.log(fieldValues))
+    // startWith([]),
+    // catchError(() => of([]))
+  ) as Observable<FieldValue[] | DateRange[]>
+
+  dateRange$ = this.fieldValues$.pipe(
+    filter(() => this.fieldType === 'dateRange'),
+    map((fieldValues) => fieldValues as DateRange[])
+  )
+
+  values$ = this.fieldValues$.pipe(
+    filter(() => this.fieldType === 'values'),
+    map((fieldValues) => fieldValues as FieldValue[])
+  )
+
+  constructor(
+    private searchFacade: SearchFacade,
+    private searchService: SearchService,
+    private fieldsService: FieldsService
+  ) {}
+
+  ngOnInit() {
+    this.fieldType = this.fieldsService.getFieldType(this.fieldName)
+  }
+
+  async removeFilterValue(fieldValue: FieldValue | DateRange) {
+    const currentFieldValues: (FieldValue | DateRange)[] = await firstValueFrom(
+      this.fieldValues$
+    )
+    const updatedFieldValues = currentFieldValues.filter(
+      (value: string | DateRange) => value !== fieldValue
+    )
+    this.fieldsService
+      .buildFiltersFromFieldValues({
+        [this.fieldName]: updatedFieldValues as FieldValue[],
+      })
+      .subscribe((filters) => this.searchService.updateFilters(filters))
```

Comment:
how about directly modifying the current filters instead of going from filters to values and then values to filters? simply deleting the corresponding key in the filters should be enough

Reply:
I must be thinking too complicated, but what would this look like? An additional method in the service or facade to delete one filter by its key? Also, date ranges don't use their value as a key in the object.

Reply:
Ok I looked into it and honestly I couldn't find a better way, so I think your approach makes sense. It ends up being verbose because we need to alternate between the "filters" model and the "values" model, but I guess this is something we cannot avoid...